StreamChatCoreTests/Unit Tests/ClientLoggerTests.swift

```diff
@@ -0,0 +1,160 @@
+                        .sendEvent(.channelDeleted, testChannel),
+                        .sendMessageAction(.init(channel: testChannel, message: testMessage, action: .init(name: "test", value: "test", style: .default, type: .button, text: "test"))),
+                        .addMembers(testMembers, testChannel),
+                        .removeMembers(testMembers, testChannel),
+                        .invite(testMembers, testChannel),
+                        .inviteAnswer(.init(channel: testChannel, accept: nil, reject: nil, message: testMessage)),
+                        .replies(testMessage, .none),
+                        .deleteMessage(testMessage),
+                        .addReaction(.angry, testMessage),
+                        .deleteReaction(.angry, testMessage),
+                        .flagMessage(testMessage),
+                        .unflagMessage(testMessage),
+                        .users(.init(filter: testFilter)),
+                        .updateUsers([testUser]),
+                        .muteUser(testUser),
+                        .unmuteUser(testUser),
+                        .flagUser(testUser),
+                        .unflagUser(testUser),
+                        .ban(.init(user: testUser, channel: testChannel, timeoutInMinutes: 0, reason: nil))]
+    }
+    
+    func testLogEndpointStrings() {
+        for endpoint in allEndpoints {
+            logger.log("Endpoint: \(String(describing: endpoint))", level: .debug)
+        }
+        
+        XCTAssertEqual(logged.count, allEndpoints.count)
+    }
+    
+    func testLogURLRequest() {
+        for endpoint in allEndpoints {
+            var urlRequest = URLRequest(url: testUrl)
+            urlRequest.httpMethod = endpoint.method.rawValue
+            urlRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+            if let body = endpoint.body {
+                let encodable = AnyEncodable(body)
+                
+                do {
+                    if let httpBody = try? JSONEncoder.defaultGzip.encode(encodable) {
+                        urlRequest.httpBody = httpBody
+                        urlRequest.addValue("gzip", forHTTPHeaderField: "Content-Encoding")
+                    } else {
+                        urlRequest.httpBody = try JSONEncoder.default.encode(encodable)
+                    }
+                } catch {
+                    XCTFail(error.localizedDescription)
+                }
+            }
+            
+            logger.log(urlRequest)
+        }
+        
+        XCTAssertFalse(logged.isEmpty)
+    }
+    
+    func testLogHeaders() {
+        let headers = [
+            "X-Stream-Client": "stream-chat-swift-client-2",
+            "X-Stream-Device": "iPhone11",
+            "X-Stream-OS": "iOS13.3",
+            "Stream-Auth-Type": "anonymous"]
+        
+        logger.log(headers: headers)
+        
+        XCTAssertFalse(logged.isEmpty)
```

Comment:
Can you check here and below logged values?

Reply:
Done